parserator_web/static/js/index.js

```diff
@@ -1,2 +1,86 @@
 /* TODO: Flesh this out to connect the form to the API and render results
    in the #address-results div. */
+
+window.onload = setSubmitBtnListener;
+
+function setSubmitBtnListener() {
+   // set up listener for a form submit event
+   const form = document.querySelector('form');
+   form.addEventListener('submit', event => {
+      event.preventDefault();
+      submitClicked();
+   });
+}
+
+function submitClicked() {
+
+   //get address text
+   const address = document.querySelector("#address").value; //should get address string
+
+   if (address.trim() == '') {
+      displayError('No input detected. Please enter an address to parse.');
+      return;
+   }
+
+   //send address to parser
+   fetch("/api/parse/?" + new URLSearchParams({address:address}).toString())
+      .then(response => {
+         response.json()
+            .then(data => {
+               if (response.ok){
+                  populateResults(data);
+               }
+               else {
+                  displayError(data['detail'])
+               }
+            });
+      })
+      .catch(err => {
+         displayError("There was a problem accessing the parser.")
+      })
+}
+
+function populateResults(data) {
+
+   // make sure results div is visible
+   setVisible("results");
+   
+   const span = document.getElementById("parse-type")
+   span.innerHTML = data['address_type']
+   
+   const table = document.querySelector('table');
+   //clear the table
+   table.innerHTML = '';
+
+   for (component in data["address_components"]) {
+      let row = table.insertRow();
+      let val = row.insertCell(0);
+      val.innerHTML = data["address_components"][component];
+      let part = row.insertCell(1);
+      part.innerHTML = component;
+   }
+}
+
+function displayError(message) {
+   const span=document.getElementById("error");
+   span.innerHTML = message;
```

Comment:
using the full error message here is a little bit too much information to share with the user. could you update this to produce a more friendly error message?

Reply:
I agree. 

I changed the error message that is sent in parserator_web/views.py to be more user-friendly. Because the message now only applies to parsing errors, I changed from catching all exceptions to only catching RepeatedLabelError. 

Since this means that any other errors that get raised will not be handled, and will return a 500 response, I also added a catch and error message for if the 'address' key is not included in the request since this falls under a 'Bad request' type of error. A regular web user should never see this because our front end should format the request correctly, but someone poking around in a console, or trying to access the parser through curl might.